server/src/test/java/io/crate/replication/logical/action/TransportCreateSubscriptionActionTest.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.replication.logical.action;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.transport.TransportService;
+import org.junit.Test;
+
+import io.crate.action.FutureActionListener;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Schemas;
+import io.crate.replication.logical.LogicalReplicationService;
+import io.crate.replication.logical.metadata.ConnectionInfo;
+import io.crate.replication.logical.metadata.RelationMetadata;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.user.User;
+import io.crate.user.UserLookup;
+
+public class TransportCreateSubscriptionActionTest extends CrateDummyClusterServiceUnitTest {
+
+    private final LogicalReplicationService logicalReplicationService = mock(LogicalReplicationService.class);
+    private final UserLookup userLookup = mock(UserLookup.class);
+    private TransportCreateSubscriptionAction transportCreateSubscriptionAction;
+
+
+    @Test
+    public void test_subscribing_to_tables_with_higher_version_raises_error() throws Exception {
+        FutureActionListener<AcknowledgedResponse, AcknowledgedResponse> responseFuture = FutureActionListener.newInstance();
+        subscribeToTablesWithVersion(responseFuture, Version.CURRENT.internalId + 10000);
+
+        assertThatThrownBy(
+            () -> responseFuture.get()
+        )
+        .hasMessageContaining("One of the published tables has version higher than subscriber's minimal node version.");
+    }
+
+    @Test
+    public void test_subscribing_to_tables_with_higher_hotfix_works() throws Exception {
+        FutureActionListener<AcknowledgedResponse, AcknowledgedResponse> responseFuture = FutureActionListener.newInstance();
+        subscribeToTablesWithVersion(responseFuture, Version.CURRENT.internalId + 1);
```

Comment:
hotfix versions are increased by 100
```suggestion
        subscribeToTablesWithVersion(responseFuture, Version.CURRENT.internalId + 100);
```

Reply:
thanks!